extralit-server/src/extralit_server/jobs/ocr_jobs.py

```diff
@@ -198,53 +166,60 @@ def run_marker(pdf_path: str, config_dict: dict[str, Any], model_dict: dict[str,
         renderer=config_parser.get_renderer(),
     )
 
-    # This should return JSONOutput because of our config
     result = converter(pdf_path)
-
-    # Verify we got JSONOutput as expected
     if not hasattr(result, "model_dump"):
         raise ValueError(f"Expected a Pydantic model with model_dump (like JSONOutput), but got {type(result)}")
-
     return result
 
 
 def parse_marker_output(result: "JSONOutput") -> dict[str, Any]:
     """
     Parse Marker JSONOutput into our application's expected layout format.
-
-    Args:
-        result: JSONOutput object from Marker
-
-    Returns:
-        A dictionary with a structured list of pages and their blocks.
     """
     layout_data = {"pages": []}
-
     if result.children:
         for page_idx, page in enumerate(result.children):
             page_data = {"page": page_idx, "blocks": []}
-
             if page.children:
                 for block in page.children:
                     block_data = {
-                        "type": block.block_type or "unknown",
-                        "bbox": block.bbox or [],
-                        "content": (block.html or "").strip(),
-                        "id": block.id or "",
-                        "score": None,  # Marker doesn't provide confidence scores
+                        "type": getattr(block, "block_type", None) or "unknown",
+                        "bbox": getattr(block, "bbox", None) or [],
+                        "content": (getattr(block, "html", None) or "").strip(),
+                        "id": getattr(block, "id", None) or "",
+                        "score": None,
                     }
                     page_data["blocks"].append(block_data)
-
             layout_data["pages"].append(page_data)
+    return layout_data
 
+
+def parse_marker_json_output(result_json: dict[str, Any]) -> dict[str, Any]:
+    """
+    Parse the JSON renderer payload returned by Modal (modal_resp['json']).
+    Mirrors Marker JSONOutput.model_dump().
+    """
+    layout_data = {"pages": []}
+    children = result_json.get("children") or []
+    for page_idx, page in enumerate(children):
+        page_data = {"page": page_idx, "blocks": []}
+        for block in page.get("children") or []:
+            block_data = {
+                "type": block.get("block_type") or "unknown",
+                "bbox": block.get("bbox") or [],
+                "content": (block.get("html") or "").strip(),
+                "id": block.get("id") or "",
+                "score": None,
+            }
+            page_data["blocks"].append(block_data)
+        layout_data["pages"].append(page_data)
     return layout_data
```

Comment:
Hey @priyankeshh, these code are handling nested dict for both function arguments and returns types, and it's better to define our own data types with pydantic.BaseModel or import a defined type from `marker`. We want to avoid using these nested dict parsing method since it's hard to maintain this code when there's no typehinting. Can you define the models at `extralit-server/src/extralit_server/api/schemas/v1/document/layout.py`? 

You can see here for an example: https://github.com/Extralit/extralit/blob/e919e0453c808c89a4e7bfa331f1542fde5c2674/extralit-server/src/extralit_server/api/schemas/v1/document/metadata.py